### DISCUSSION TITLE
Add Core\Form\ConfigurationDataProvider

### DISCUSSION THREAD

**Discussion by tom-combet** [PrestaShop/PrestaShop#33426] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

As a developer, I want to use a generic form data provider for settings forms so that I can use it in my form data provider service with my DataConfigurationInterface class instead of writing a new one.

### Describe the solution you'd like

Add `Core\Form\ConfigurationDataProvider` class which will use the provided DataConfigurationInterface to get and update configuration values.

### Additional context

![image](https://user-images.githubusercontent.com/48719560/162566220-426774cc-2fc3-4340-87e2-4ff4aecd147b.png)

As one can see, logic is pretty much the same for all settings forms because validation is handled in Form Types now.

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [ ] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by hibatallahAouadni** [comment]
Hello @tom-combet 

Thank you for your suggestion. The Developer Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you
